Fix : test 코드의 errors 응답 수정

200wonman opened the pull request:
Resolves #{이슈-번호}


# 해결하려는 문제가 무엇인가요?

- test 코드의 errors 응답 수정

# 어떻게 해결했나요?

- 포멧에 맞추어 응답이 보이도록 수정했습니다.

# 참고

- request null값과 유저쪽 에러는 처리 방법에 대해  진행중입니다.

## 참고 자료

-

---

# RCA 룰

r: 꼭 반영해 주세요. 적극적으로 고려해 주세요. (Request changes)  
c: 웬만하면 반영해 주세요. (Comment)  
a: 반영해도 좋고 넘어가도 좋습니다. 그냥 사소한 의견입니다. (Approve)

Whale0928 committed:
fix: global response error 처리 방식 변경에 따른 테스트 수정 및 단일 에러 반환을 리스트로 변경

Whale0928 committed:
fix: global response error 처리 방식 변경에 따른 테스트 수정 및 단일 에러 반환을 리스트로 변경

200wonman committed:
test : 글로벌 응답 메세지 포멧변환으로 인한 테스트 수정중

200wonman committed:
fix : request error message & unitTest / integrationTest modify

Whale0928 commented on src/main/java/app/bottlenote/global/exception/custom/code/ValidExceptionCode.java:
```diff
@@ -14,6 +14,10 @@ public enum ValidExceptionCode implements ExceptionCode {
 	UNKNOWN_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "알 수 없는 오류가 추가적인 문의가 필요합니다."),
 	JWT_TOKEN_EXCEPTION(HttpStatus.UNAUTHORIZED, "JWT 토큰 관련 예외가 발생했습니다."),
 	AWS_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "AWS 관련 오류가 발생했습니다. infra 팀에 문의해주세요."),
+	REQUIRED_USER_ID(HttpStatus.BAD_REQUEST, "유저 아이디 필수입니다."),
```
이게 Common에 있을건 아니죠?
그리고 유저 아이디 필수입니다 -> 유저 아이디는 필수입니다

Whale0928 commented on src/main/java/app/bottlenote/global/exception/custom/code/ValidExceptionCode.java:
```diff
@@ -14,6 +14,10 @@ public enum ValidExceptionCode implements ExceptionCode {
 	JWT_TOKEN_EXCEPTION(HttpStatus.UNAUTHORIZED, "JWT 토큰 관련 예외가 발생했습니다."),
 	AWS_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "AWS 관련 오류가 발생했습니다. infra 팀에 문의해주세요."),
+	REQUIRED_USER_ID(HttpStatus.BAD_REQUEST, "유저 아이디 필수입니다."),
+	NOT_BLANK(HttpStatus.BAD_REQUEST, "공백은 입력할 수 없습니다."),
```
이렇게 입력하면 어떤 값에서 공백이 발생했는지 어떻게 알까요? 
개별적으로 만들어져야합니다.

Whale0928 commented on src/main/java/app/bottlenote/global/exception/custom/code/ValidExceptionCode.java:
```diff
@@ -14,6 +14,10 @@ public enum ValidExceptionCode implements ExceptionCode {
 	AWS_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "AWS 관련 오류가 발생했습니다. infra 팀에 문의해주세요."),
+	REQUIRED_USER_ID(HttpStatus.BAD_REQUEST, "유저 아이디 필수입니다."),
+	NOT_BLANK(HttpStatus.BAD_REQUEST, "공백은 입력할 수 없습니다."),
+	NOT_NULL(HttpStatus.BAD_REQUEST, "null은 입력할 수 없습니다."),
```
위와 동일

Whale0928 commented on src/main/java/app/bottlenote/global/exception/custom/code/ValidExceptionCode.java:
```diff
@@ -14,6 +14,10 @@ public enum ValidExceptionCode implements ExceptionCode {
+	REQUIRED_USER_ID(HttpStatus.BAD_REQUEST, "유저 아이디 필수입니다."),
+	NOT_BLANK(HttpStatus.BAD_REQUEST, "공백은 입력할 수 없습니다."),
+	NOT_NULL(HttpStatus.BAD_REQUEST, "null은 입력할 수 없습니다."),
+	NOT_EMPTY(HttpStatus.BAD_REQUEST, "빈 값은 입력할 수 없습니다."),
```
만약 하나의 값으로만 이루어진 필드가 있다면 유요하지 위 셋은 지우지 말고 주석으로 설명을 추가해두세요. 다른 필드가 전혀 없을 경우 사용해도 된다고.

Whale0928 commented on src/main/java/app/bottlenote/global/exception/custom/code/ValidExceptionCode.java:
```diff
@@ -22,8 +26,34 @@ public enum ValidExceptionCode implements ExceptionCode {
+
+	// LIKE
+	REVIEWID_ID_REQUIRED(HttpStatus.BAD_REQUEST, "reviewId(식별자)는 필수입니다."),
+	STATUS_IS_REQUIRED(HttpStatus.BAD_REQUEST, "status는 필수입니다."),
```
status는 굉장히 많으니 좀더 정확하게 어느 status인지 명시 필요

Whale0928 commented on src/test/java/app/bottlenote/follow/controller/FollowUpdateControllerTest.java:
```diff
@@ -165,12 +169,12 @@ void test_4() throws Exception {
-		resultActions.andExpect(jsonPath("$.code").value(400));
-		resultActions.andExpect(jsonPath("$.errors.message").value("팔로우할 대상을 찾을 수 없습니다."));
+		resultActions.andExpect(jsonPath("$.code").value(404));
+		resultActions.andExpect(jsonPath("$.errors[0].message").value("팔로우할 대상을 찾을 수 없습니다."));
```
이건 왜 하드코딩? enum에 정의되있으니 굳이 하드코딩할 필요가 없겠네요

Whale0928 commented on src/test/java/app/bottlenote/user/controller/OauthControllerTest.java:
```diff
@@ -179,10 +191,14 @@ void user_reissue_fail_when_refreshToken_is_not_invalid() throws Exception {
+			.andExpect(jsonPath("$.errors[0].code").value(String.valueOf(error.code())))
+			.andExpect(jsonPath("$.errors[0].status").value(error.status().name()))
+			.andExpect(jsonPath("$.errors[0].message").value(error.message()));
+//		[{"code":"INVALID_REFRESH_TOKEN","status":"UNAUTHORIZED","message":"유효하지 않은 리프레쉬 토큰입니다."}]
```
무의미한 주석이면 제거

Whale0928 reviewed (changes requested):
확인하고 수정하죠

200wonman committed:
refactor : modifyTestCodeAndValidExceptionCode

Whale0928 committed:
refactor: add review reply status to response object (#228)

# 해결하려는 문제가 무엇인가요?

- 리뷰 댓글 조회 시 상태 값을 함께 전달하도록 수정.



---

# RCA 룰

r: 꼭 반영해 주세요. 적극적으로 고려해 주세요. (Request changes)  
c: 웬만하면 반영해 주세요. (Comment)  
a: 반영해도 좋고 넘어가도 좋습니다. 그냥 사소한 의견입니다. (Approve)

200wonman committed:
refactor: fix testcode and ValidExceptionCode
